apps/web/src/components/settings/index.tsx

```diff
@@ -105,6 +106,7 @@ export function Settings({ profileData, username }: { profileData: FormSchema; u
                       defaultValue={val.url}
                       placeholder="Link to social profile"
                       {...register(`userLinks.${i}.url`)}
+                      onChange={() => setFocus(`userLinks.${i}.url`, { shouldSelect: true })}
```

Comment:
you shouldnt have to setfocus on a keystroke

Reply:
- Tried wrapping it with Controller component from react-hook-form , didn't worked their 
- Tried changing the key of FormItem component for not causing re-render , it didn't help either 
Can you please suggest something